### PR TITLE
[cli] change patch-inquirer-legacy to patch-inquirer

### PR DESCRIPTION
**Pull request by techoi**
### Related Issues

> Fixes to unify inquirer patch for output style and override chalk.
> Related to https://github.com/vercel/vercel/pull/8826 https://github.com/vercel/vercel/pull/8826#discussion_r1023228964

#### Tests

- [x] The code changed/added as part of this PR has been covered with tests
- [x] All tests pass locally with `yarn test-unit`

#### Code Review

- [ ] This PR has a concise title and thorough description useful to a reviewer
- [ ] Issue from task tracker has a link to this PR

### CODE REVIEW

**Comment by techoi** [comment]
@EndangeredMassa Could you approve this PR to complete no-color mode issue?

**Comment by EndangeredMassa** [comment]
CI tests don't run for external contributors. I ran these tests and found a couple failures. Can you try running these tests locally and fix them?

```
FAIL test/unit/commands/login.test.ts (11.377 s)
  login
    interactive
      ✕ should allow the `--no-color` flag (3008 ms)
      with NO_COLOR="1" env var
        ✕ should remove emoji the `NO_COLOR` env var with 1 (3007 ms)
      with FORCE_COLOR="0" env var
        ✕ should remove emoji the `FORCE_COLOR` env var with 0 (3007 ms)
```

They all fail with:

```
Timed out waiting 3000 ms for output
```

on line:

```
await expect(client.stderr).toOutput(`> Log in to Vercel`);
```

**Comment by techoi** [comment]
@EndangeredMassa Thank you for commnet I fix it in c8e0ea9

File: packages/cli/src/util/input/patch-inquirer.ts [hunk 1]
```diff
@@ -84,7 +84,10 @@ function listRender(choices: (Choice | Separator)[], pointer: number) {
 
     let isSelected = i - separatorOffset === pointer;
     let line = (isSelected ? '● ' : '○ ') + choice.name;
-    line = chalk.cyan(line);
+
+    if (isSelected) {
+      line = chalk.cyan(line);
+    }
```
Comment (EndangeredMassa) [inline packages/cli/src/util/input/patch-inquirer.ts:90]: This is a great change, thanks!

File: packages/cli/src/util/input/patch-inquirer-legacy.ts [hunk 2]
```diff
@@ -1,21 +0,0 @@
-import chalk from 'chalk';
-import inquirer from 'inquirer';
-import Prompt from 'inquirer/lib/prompts/base';
-
-// Here we patch inquirer to use a `>` instead of the ugly green `?`
-
-/* eslint-disable no-multiple-empty-lines, no-var, no-undef, no-eq-null, eqeqeq, semi */
-const getQuestion = function (this: Prompt) {
-  var message = `${chalk.bold(`> ${this.opt.message}`)} `;
```
Comment (EndangeredMassa) [inline packages/cli/src/util/input/patch-inquirer-legacy.ts:9]: Before:

CleanShot 2023-02-28 at 10 57 54@2x

After:

CleanShot 2023-02-28 at 10 58 09@2x

**Review by EndangeredMassa — APPROVED** [review]

**Review by cb1kenobi — APPROVED** [review]

**Comment by EndangeredMassa** [comment]
I confirmed these tests pass. Merging.

**Pull request merged**